flocker/provision/_aws.py

```diff
@@ -220,33 +222,34 @@ def create_node(self, name, distribution,
                 filters={'name': IMAGE_NAMES[distribution]},
             )
 
-            with start_action(
-                action_type=u"flocker:provision:aws:create_node:run_instances",
-            ) as context:
-                reservation = self._connection.run_instances(
-                    images[0].id,
-                    key_name=self._keyname,
-                    instance_type=size,
-                    security_groups=self._security_groups,
-                    block_device_map=diskmap,
-                    placement=self._zone,
-                    # On some operating systems, a tty is requried for sudo.
-                    # Since AWS systems have a non-root user as the login,
-                    # disable this, so we can use sudo with conch.
-                    user_data=dedent("""\
-                        #!/bin/sh
-                        sed -i '/Defaults *requiretty/d' /etc/sudoers
-                        """),
-                )
-
-                instance = reservation.instances[0]
-                context.add_success_fields(instance_id=instance.id)
-
-            self._connection.create_tags([instance.id], metadata)
-
-            # Display state as instance starts up, to keep user informed that
-            # things are happening.
-            _wait_until_running(instance)
+            action_type = u"flocker:provision:aws:create_node:run_instances"
+            while True:
+                with start_action(
+                    action_type=action_type,
+                ) as context:
+                    reservation = self._connection.run_instances(
+                        images[0].id,
+                        key_name=self._keyname,
+                        instance_type=size,
+                        security_groups=self._security_groups,
+                        block_device_map=diskmap,
+                        placement=self._zone,
+                        # On some operating systems a tty is requried for sudo.
+                        # Since AWS systems have a non-root user as the login,
+                        # disable this, so we can use sudo with conch.
+                        user_data=dedent("""\
+                            #!/bin/sh
+                            sed -i '/Defaults *requiretty/d' /etc/sudoers
+                            """),
+                    )
+
+                    instance = reservation.instances[0]
+                    context.add_success_fields(instance_id=instance.id)
+
+                self._connection.create_tags([instance.id], metadata)
+
+                if _wait_until_running(instance):
+                    break
```

Comment:
Probably wouldn't hurt to make the body of this loop a separate function now to help with the readability of `create_node`.

Also, I'd probably put a pretty small retry limit on this.  If something totally weird is going on, I wouldn't like it if this code ends up booting 50 or 100 instances before I realize what's happening and kill it.  Also, worse, buildbot isn't smart and could take up to an hour to get bored.  That could be a lot of instances.